src/wallet/rpcwallet.cpp

```diff
@@ -2032,9 +2092,15 @@ UniValue getwalletinfo(const UniValue& params, bool fHelp)
     if (pwalletMain->IsCrypted())
         obj.pushKV("unlocked_until", nWalletUnlockTime);
     obj.pushKV("paytxfee",      ValueFromAmount(payTxFee.GetFeePerK()));
-    uint256 seedFp = pwalletMain->GetHDChain().seedFp;
-    if (!seedFp.IsNull())
-         obj.pushKV("seedfp", seedFp.GetHex());
+    auto mnemonicChain = pwalletMain->GetMnemonicHDChain();
+    if (mnemonicChain.has_value())
+         obj.pushKV("mnemonic_seedfp", mnemonicChain.value().GetSeedFingerprint().GetHex());
+    // TODO: do we really need to return the legacy seed fingerprint if we're
+    // no longer using it to generate any new keys? What do people actually use
+    // the fingerprint for?
+    auto legacySeed = pwalletMain->GetLegacyHDSeed();
+    if (legacySeed.has_value())
+        obj.pushKV("legacy_seedfp", legacySeed.value().Fingerprint().GetHex());
```

Comment:
We definitely shouldn't alter the name of the legacy key, as that's what breaks backwards-compatibility.
```suggestion
        obj.pushKV("seedfp", legacySeed.value().Fingerprint().GetHex());
```
If we want to deprecate this field, we can do so later (once we figure out a better process for handling those kinds of partial deprecations). For now, we can mark this field as "legacy" in the help text.

Reply:
In this case I opted to *intentionally* break backwards compatibility, because any assumptions that code using the `seedfp` field is making are likely to be invalidated by the switch to the mnemonic seed. By breaking compatibility here, we force users to take those changes into account, so I recommend leaving this as I wrote it.

Reply:
If it's an intentional compatibility break, it should be documented (both in the help text of the RPC method, and in the release notes).